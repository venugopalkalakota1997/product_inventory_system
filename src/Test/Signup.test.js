import React from 'react';
import { MemoryRouter as Router, withRouter } from 'react-router-dom'
import {mount} from 'enzyme';
import {shallow} from 'enzyme';
import Signup from '../compoents/Signup';
import {configure} from 'enzyme'
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter:new Adapter()});

it('Render',()=>{
  const wrapper=shallow(<Signup />);
  const h=wrapper.find("h2");
  const result=h.text();

  expect(result).toBe("Signup Component");
})

test('Login', () => {
    expect(true).toBeTruthy();
  });

it('Submit Fucntion',()=>{
  const onSubmitFun=jest.fn();
  const wrapper=mount(<form onSubmit={onSubmitFun}></form>)
  const form=wrapper.find('form');
  form.simulate('submit');
  expect(onSubmitFun).toHaveBeenCalledTimes(1);
});
it('username field',()=>{
  const wrapper=mount(<input type="text" name="username"/>);
  const input=wrapper.find('input');
  expect(input).toHaveLength(1);
  expect(input.prop('type')).toEqual('text');
  expect(input.prop('name')).toEqual('username');
})
it('Password field',()=>{
  const wrapper=mount(<input type="password" name="password"/>);
  const input=wrapper.find('input');
  expect(input).toHaveLength(1);
  expect(input.prop('type')).toEqual('password');
  expect(input.prop('name')).toEqual('password');
})
it('Button field',()=>{
  const wrapper=mount(<button type="submit">Login</button>);
  const input=wrapper.find('button');
  const result=input.text();
  expect(result).toBe('Login')
  expect(input.prop('type')).toEqual('submit');
})
test('mathes Snapshot',()=>{
  const tree=renderer.create(<Router><Signup/></Router>).toJSON()
  expect(tree).toMatchSnapshot();
})