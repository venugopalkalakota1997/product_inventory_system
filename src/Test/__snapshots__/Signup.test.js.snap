// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mathes Snapshot 1`] = `
<div>
  <form
    className="form"
  >
    <h2>
      Signup Component
    </h2>
    <p>
      Name 
      <span
        style={
          Object {
            "color": "red",
          }
        }
      >
         *
      </span>
    </p>
    <input
      className="input"
      id="username"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
    />
    <p
      className="error"
    >
      
    </p>
    <p>
      Contact Number 
      <span
        style={
          Object {
            "color": "red",
          }
        }
      >
         *
      </span>
    </p>
    <input
      className="input"
      id="contactNumber"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
    />
    <p
      className="error"
    >
      
    </p>
    <p>
      Password
      <span
        style={
          Object {
            "color": "red",
          }
        }
      >
         *
      </span>
       
    </p>
    <input
      className="input"
      id="password"
      onBlur={[Function]}
      onChange={[Function]}
      type="password"
    />
    <p
      className="error"
    >
      
    </p>
    <p>
      Confirm Password 
      <span
        style={
          Object {
            "color": "red",
          }
        }
      >
         *
      </span>
    </p>
    <input
      className="input"
      id="confirmPassword"
      onBlur={[Function]}
      onChange={[Function]}
      type="password"
    />
    <br />
    <p
      className="error"
    >
      
    </p>
    <br />
    <button
      className="button"
      disabled={true}
      onClick={[Function]}
      type="submit"
    >
      Signup
    </button>
  </form>
</div>
`;
